### PR TITLE
Chore: Add CSS sourcemaps

### LINKED ISSUE #642: CSS source maps in Storybook
When developing in Storybook and inspecting elements, all that's ever shown is `inline:*` for the location of the CSS. This isn't very useful:

Screen Shot 2020-04-22 at 2 26 52 PM

Ideally we could embed source maps in our output when running locally, allowing us to debug more easily. But after spending around 45 minutes attempting to troubleshoot, I was unsuccessful.

Things I tried:

- Adding `{ options: { sourceMap: true } }` to the `css-loader`, `postcss-loader` and `sass-loader` rules in our configuration.
- Removing the `postcss-loader` step entirely (and checking out an old commit) to make sure that step wasn't messing with anything (it does not seem to be).
- Changing the `config.devtool` option in `webpackFinal` (did not seem to have any perceivable effect).

I feel like I'm missing something silly, but I've spent as much time as I'm willing to at the moment.

### PR DESCRIPTION
**Pull request by Josh68**
## Overview

Adds configuration to bubble up CSS sourcemaps via various Webpack loader configurations.

Swaps the `style-loader` for getting CSS into pages with the `mini-css-extract-plugin` and uses an alternate `devtool` strategy for development and production source maps. This appears to be a combination that works across browsers and in both development and the build.

See https://github.com/webpack-contrib/style-loader/issues/303#issuecomment-581168870
and https://webpack.js.org/plugins/mini-css-extract-plugin/

## Screenshots

From preview theme, Embed -> Image

image

## Testing

1. [Preview theme](https://deploy-preview-643--cloudfour-patterns.netlify.app)
1. Inspect CSS for various components, objects, etc, to ensure the source is identified in its original file and correct line number(s)

---

- Fixes #642

/CC @tylersticka

### CODE REVIEW

**Comment by tylersticka** [comment]
@Josh68 Apologies for peeking at a draft PR, but I thought I'd mention that this seems to fix things in Chrome and Safari but not Firefox?

Possibly related: https://github.com/webpack-contrib/style-loader/issues/303

**Comment by Josh68** [comment]
> @Josh68 Apologies for peeking at a draft PR, but I thought I'd mention that this seems to fix things in Chrome and Safari but not Firefox?
> 
> Possibly related: [webpack-contrib/style-loader#303](https://github.com/webpack-contrib/style-loader/issues/303)

Stickier issue than I thought. Trying out some other suggested options.

**Comment by Josh68** [comment]
@tylersticka, I think this combination of updates works out. I'm running FF on windows as I type this, and have been able to get to the correct source lines in both development and the preview build. Unfortunately, this isn't as simple as it ought to be, currently. 🤞

**Comment by tylersticka** [comment]
I'm noticing a lot of Terminal noise related to the `mini-css-extract-plugin`. Is that expected? Is there any way to quiet it down?

Screen Shot 2020-04-23 at 9 17 00 AM

**Comment by Josh68** [comment]
> I'm noticing a lot of Terminal noise related to the `mini-css-extract-plugin`. Is that expected? Is there any way to quiet it down?

I'll have a look

**Comment by Josh68** [comment]
> > I'm noticing a lot of Terminal noise related to the `mini-css-extract-plugin`. Is that expected? Is there any way to quiet it down?
> 
> I'll have a look

@tylersticka, if I just fire this up locally via `npm run storybook` and change a line in a random `scss` file, I don't see any different output than I did previously, and it's minimal

```
webpack building...
webpack built a805db16e4164b830a93 in 959ms
```

If I run `npm run build-storybook`, there's no verbose output.

What are you doing when you see this output?

**Comment by Josh68** [comment]
I am going to briefly try out reverting to use of `style-loader` and trying out the [source-map-dev-tool-plugin](https://webpack.js.org/plugins/source-map-dev-tool-plugin/), with minimal or no configuration (and setting `config.devtool = false`, since these cannot be used together). Just want to see how FF handles this in the built preview.

**Comment by Josh68** [comment]
@tylersticka, whenever you have a chance, give me a bit more detail on when you encountered the output shown in https://github.com/cloudfour/cloudfour.com-patterns/pull/643#issuecomment-618494558. I tried out some other options, but timeboxed the effort. What I have now is working as intended, AFAICT, and I also haven't seen what your reported (but certainly may be missing something). Thanks.

**Comment by tylersticka** [comment]
@Josh68 I can't reproduce it now. Not sure what happened. 🤔 

Oh well! Reviewing now...

File: .storybook/main.js [hunk 1]
```diff
@@ -23,27 +25,48 @@ module.exports = {
       return data;
     });
 
+    /**
+     * For development, leave the default 'cheap-module-source-map', as it's faster and works
+     * For the build, using the default does not work correctly, but this option appears to
```
Comment (tylersticka) [inline .storybook/main.js:30]: Nit: Should these sentences end in periods? At first glance I thought this was a run-on sentence.

File: .storybook/main.js [hunk 2]
```diff
@@ -23,27 +25,48 @@ module.exports = {
...
+    }
+
     // Push new rules
     config.module.rules.push(
       {
         test: /\.s[ca]ss$/,
         use: [
-          'style-loader',
+          {
+            loader: MiniCssExtractPlugin.loader, // @see https://github.com/webpack-contrib/style-loader/issues/303#issuecomment-581168870
```
Comment (tylersticka) [inline .storybook/main.js:43]: Nit: Can we put this `@see` on its own line so it's easier to find?

File: package.json [hunk 3]
```diff
@@ -65,13 +65,15 @@
...
     "ltx": "2.9.2",
+    "mini-css-extract-plugin": "^0.9.0",
     "npm-run-all": "4.1.5",
     "postcss-inline-svg": "4.1.0",
     "postcss-loader": "3.0.0",
     "prettier": "2.0.5",
     "remark-preset-lint-recommended": "4.0.0",
     "remark-preset-prettier": "0.4.0",
     "require-dir": "1.2.0",
+    "resolve-url-loader": "^3.1.1",
```
Comment (tylersticka) [inline package.json:76]: I don't see this referenced in our code, is it still required or is it an artifact of an earlier exploration?

Comment (Josh68) [reply package.json:76]: > artifact of an earlier exploration?

You got it. Removing

**Comment by Josh68** [comment]
@tylersticka, all feedback addressed, I hope. 🙇‍♂️ 

Redeployment working as expected.

**Pull request merged**